### DISCUSSION TITLE
In-Memory, Kubernetes supports and future of helyim

### DISCUSSION THREAD

**Discussion by Oyelowo** [helyim/helyim#126] [General]
Great job on building this 👍 

- What are the plans for this project? Is it mainly for learning purpose or to build to be on-par with seaweedfs  or even branch out to be independently more capable?

- Would this support  in-memory for testing? 
 My allure to using rust for most of my infrastructure is so I can use same APIs in testing and local development as I will use in production without having to mock anything anywhere. Exposing the ability to do some in-memory or local-file writes/reads would be nice

- Are you considering potential support for kubernetes with a helm chart?

- Would you consider having a roadmap and future plans proposals which can be linked in the readme? This could also include list of things that are implemented, planned to be implemented etc

**Comment by iamazy** [comment]
@Oyelowo Hi, sorry for the late reply.
- This project is mainly for learning purposes, but I hope to give it more abilities, although helyim is too immature to compared with seaweedfs right now. I will continue to develop helyim and stabilize it, and I hope it can eventually be used in production environments. I'm currently exploring how to make helyim support `io_uring` and support `kernel bypass` by integrating [demikernel](https://github.com/microsoft/demikernel).

- helyim supports testing in memory. You can easily set up helyim on your computer according to the tutorial in the [README](https://github.com/helyim/helyim/blob/main/README.md)

- helyim has plans to support k8s with helm chart

- helyim's recent roadmap is to implement filer to support s3, fuse etc.
